naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2433,4 +2454,43 @@ impl Parser {
         }
         Ok(brace_nesting_level + 1)
     }
+
+    fn diagnostic_control<'a>(&self, lexer: &mut Lexer<'a>) -> Result<(), Error<'a>> {
+        // TODO(maybe): self.push_rule_span(Rule::???, lexer)
+        // TODO: I see `*_argument` helpers, but they're not used here. Guessing it's
+        // more valuable to make something easy to validate against the WGSL spec.?
```

Comment:
Wanted to ping @teoxoy and/or @jimblandy on this question about `Lexer` usage:

```
        // TODO: I see `*_argument` helpers, but they're not used here. Guessing it's
        // more valuable to make something easy to validate against the WGSL spec.?
```

Reply:
Honestly, I think when I see

```rust
self.expect(Token::Paren('('))?;
```

it's going to take me less time to understand and predict that code's behavior than if I see:

```rust
self.open_arguments()?;
```

I'd want to go check the definition of `open_arguments` to see if it was doing something unexpected or tricky. Neither is the case here, so I don't think `open_arguments` is providing net value to readers of Naga's WGSL front end.